doi-transformer/src/main/java/no/unit/nva/doi/transformer/CrossRefConverter.java

```diff
@@ -165,46 +177,120 @@ private Range extractPages(CrossRefDocument document) {
-        } else {
-            throw new IllegalArgumentException(String.format(UNRECOGNIZED_TYPE_MESSAGE, document.getType()));
+        if (nonNull(crossrefType)) {
+            PublicationType publicationType = crossrefType.getPublicationType();
+            if (nonNull(publicationType)) {
+                if (publicationType.equals(PublicationType.JOURNAL_CONTENT)) {
+                    return createJournalContext(document);
+                } else if (publicationType.equals(PublicationType.BOOK)) {
+                    return createBookContext(document);
+                }
+            } else {
+                throw new IllegalArgumentException(String.format(UNRECOGNIZED_TYPE_MESSAGE, document.getType()));
+            }
         }
+        throw new IllegalArgumentException(MISSING_CROSSREF_TYPE_IN_DOCUMENT);
+    }
+
+    private Book createBookContext(CrossRefDocument document) throws InvalidIsbnException {
+        return new Book.Builder()
+                .withLevel(null)
+                .withOpenAccess(false)
+                .withPeerReviewed(hasReviews(document))
+                .withSeriesTitle(extractSeriesTitle(document))
+                .withPublisher(extractPublisherName(document))
+                .withUrl(extractFulltextLink(document))
+                .withIsbnList(extractIsbn(document))
+                .build();
+    }
+
+    private Journal createJournalContext(CrossRefDocument document) throws InvalidIssnException {
+        // TODO actually call the Channel Register API and get the relevant details
+        return new Journal.Builder()
+                .withLevel(null)
+                .withTitle(extractJournalTitle(document))
+                .withOnlineIssn(extractOnlineIssn(document))
+                .withPrintIssn(extractPrintIssn(document))
+                .withOpenAccess(false)
+                .withPeerReviewed(false)
+                .build();
+    }
+
+    private boolean hasReviews(CrossRefDocument document) {
+        return nonNull(document.getReview());
+    }
+
+    private List<String> extractIsbn(CrossRefDocument document) {
+        List<Isxn> isbns = document.getIsbnType();
+        if (isNull(isbns) || isbns.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return isbns.stream()
+                .map(Isxn::getValue)
+                .collect(Collectors.toList());
+
+    }
+
+    private String extractSeriesTitle(CrossRefDocument document) {
+        if (isNull(document.getContainerTitle()) || document.getContainerTitle().isEmpty()) {
+            return null;
+        }
+        return document.getContainerTitle().stream()
+                .findFirst()
+                .orElse(null);
+    }
+
+    private String extractPublisherName(CrossRefDocument document) {
+        return document.getPublisher();
+    }
+
+    private Organization extractAndCreatePublisher(CrossRefDocument document) {
+        if (isNotEmpty(document.getPublisher())) {
```

Comment:
return (isNotEmpty(document.getPublisher())) 
                ? new Organization.Builder().withLabels(Map.of("name", document.getPublisher())).build() 
                : null;

Reply:
Terrnary..